node/consensus/leader.go

```diff
@@ -183,6 +187,7 @@ func (ce *ConsensusEngine) processVotes(ctx context.Context) error {
 		// start the next round
 		ce.nextState()
 
+		ce.wg.Add(1)
```

Comment:
This block below is one reason why I like to put the `wg.Done` always in an anonymous func.  The done is in `startNewRound`, but it's also needed in the `<-ctx.Done` return case.

Reply:
Yeah, I removed startNewRounds from a go rountine, now we only have 1 go routine to manage which is the event loop